cli/config/config.go

```diff
@@ -100,29 +106,36 @@ func load(configDir string) (*configfile.ConfigFile, error) {
 	file, err := os.Open(filename)
 	if err != nil {
 		if os.IsNotExist(err) {
-			//
-			// if file is there but we can't stat it for any reason other
-			// than it doesn't exist then stop
+			// It is OK for no configuration file to be present, in which
+			// case we return a default struct.
 			return configFile, nil
 		}
-		// if file is there but we can't stat it for any reason other
-		// than it doesn't exist then stop
-		return configFile, nil
+		// Any other error happening when failing to read the file must be returned.
+		return configFile, errors.Wrap(err, "loading config file")
 	}
 	defer file.Close()
 	err = configFile.LoadFromReader(file)
 	if err != nil {
-		err = errors.Wrap(err, filename)
+		err = errors.Wrapf(err, "loading config file: %s: ", filename)
 	}
 	return configFile, err
 }
 
 // LoadDefaultConfigFile attempts to load the default config file and returns
-// an initialized ConfigFile struct if none is found.
+// an initialized ConfigFile struct if none is found or when failing to load
```

Comment:
I'm not sure about godoc best practices when returning pointers to structs, should we specify that here as well or only in the function signature?

Reply:
Yeah, I left it out for now; I think we should consider deprecating this function, or at least provide an alternative that does return the error.

Reply:
I tried to slightly rephrase the GoDoc to make it more clear that it returns a reference, and that it initialises a default struct if it fails to load / find a config-file.